pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,232 @@
+	var forStorageClass, openEBSImage, openEBSNode, biggerThanString string
+	var biggerThanBytes int64
+	var clientSet kubernetes.Interface
+	var rookClientSet rookcli.Interface
+	var selectedClass *storagev1.StorageClass
+
+	cmd := &cobra.Command{
+		Use:   "check-free-disk-space",
+		Short: "List and analyse the available disk space for a given Storage Class.",
+		Example: fmt.Sprintf(""+
+			"In the following examples 'openebs' is the name of a storage class backed by the %s storage provisioner while 'rook' is the name of a storage\n"+
+			"class backed by the %s or %s storage provisioners.\n\n"+
+			"# reports the available space in rook\n"+
+			"kurl cluster check-free-disk-space --storageclass rook\n\n"+
+			"# reports the available space in all nodes\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs\n\n"+
+			"# checks if there is 10G available on node node0\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs --openebs-node-name node0 --openebs-image ubuntu:latest --bigger-than 10G\n\n"+
+			"# checks if there is 10G available on all nodes in the cluster\n"+
+			"kurl cluster check-free-disk-space --storageclass openebs --bigger-than 10G\n"+
+			"# checks if there is 20G available in the cluster on the default storage class\n"+
+			"kurl cluster check-free-disk-space --bigger-than 20G\n",
+			openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner,
+		),
+		Long: fmt.Sprintf(""+
+			"This program returns the amount of free disk space (in bytes) for a given storage class or compares if there is enough space to hold a certain\n"+
+			"amount of data (--bigger-than). For OpenEBS, when no --bigger-than flag is provided, this program returns a list of nodes and their respective\n"+
+			"free space, while for Rook storage only the available space is returned. When --bigger-than flag is used this program sets the exit code to zero\n"+
+			"if the space available is bigger than the quantity provided. For OpenEBS, if no node has been provided through --openebs-node-name the exit code\n"+
+			"will be zero only if all nodes free space are bigger than the provided quantity (see Examples section for more details).\n\n"+
+			"Supports the following storage provisioners: %s, %s, %s",
+			openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner,
+		),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			k8sConfig, err := config.GetConfig()
+			if err != nil {
+				return fmt.Errorf("failed to read kubernetes configuration: %w", err)
+			}
+
+			clientSet, err = kubernetes.NewForConfig(k8sConfig)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %w", err)
+			}
+
+			rookClientSet, err = rookcli.NewForConfig(k8sConfig)
+			if err != nil {
+				return fmt.Errorf("failed to create rook client: %w", err)
+			}
+
+			if selectedClass, err = getStorageClassByName(cmd.Context(), clientSet, forStorageClass); err != nil {
```

Comment:
We are just checking if the value is or not supported above.
Ideally we do like:
a) First check all inputs ( return an error if something fails )
b) After we check the inputs we run the command. In this case we can return an error here before to check if the values are or nor supported that seems a little strange.